File: Practica2/Linea.py
class Punto:
    def __init__(slef, x, y):
        slef.x = x
        slef.y = y

    def __str__(self):
        return ("{}, {}").format(self.x, self.y)


class Linea:
    def __init__(self, p1, p2):
        self.p1 = p1
        self.p2 = p2

    def __str__(self):
        return ("{}, {}").format(self.p1, self.p2)

    def dibujarLinea(self):
        print("dibujar Linea", self)


if __name__ == "__main__":
    punto1 = Punto(0, 0)
    punto2 = Punto(5, 5)
    linea = Linea(punto1, punto2)

    print(linea)
    linea.dibujarLinea()

File: Practica2/Linea.java
package Practica2;
public class Linea {
    public static class Punto {
        float x, y;

        Punto(float a, float b) {
            x = a;
            y = b;
        }

        float[] coord_cartesianas() {
            return new float[]{x, y};
        }

        double[] coord_polares() {
            double r = Math.sqrt(x * x + y * y);
            double theta = Math.atan2(y, x);
            return new double[]{r, theta};
        }
        @Override
        public String toString() {
            return x + ", " + y;
        }
    }

    Punto p1, p2;

    Linea(Punto punto1, Punto punto2) {
        p1 = punto1;
        p2 = punto2;
    }
    @Override
    public String toString() {
        return "Linea = " + p1 + " Punto " + p2;
    }

    void dibujaLinea() {
        System.out.println("Dibujando una línea desde " + p1 + " hasta " + p2);
    }

    public static void main(String[] args) {
        Punto p1 = new Punto(1, 2);
        Punto p2 = new Punto(3, 4);
        Linea linea = new Linea(p1, p2);

        System.out.println(linea);
        linea.dibujaLinea();
        float[] cartesianCoords = p1.coord_cartesianas();
        System.out.println("Punto p1 coordenadas cartesianas: (" + cartesianCoords[0] + ", " + cartesianCoords[1] + ")");
        double[] polarCoords = p1.coord_polares();
        System.out.println("Punto p1 coordenadas polares: (r = " + polarCoords[0] + ", O = " + polarCoords[1] + ")");
    }   
}
